atom/browser/web_dialog_helper.cc

```diff
@@ -56,10 +108,13 @@ void WebDialogHelper::RunFileChooser(content::WebContents* web_contents,
     }
 
     std::vector<base::FilePath> paths;
+    auto browser_context = AtomBrowserMainParts::Get()->browser_context();
```

Comment:
> The current implementation relies on the PrefService in brightray::BrowserContext, which is aligned with Chromium's. And content::BrowserContext doesn't has such interface.

We can get it from current window.

> I know current implementation will cause <input> using the last used directory even in different WebContents, but this behavior is same with Chromium's(Chromium will share this last used directory to all WebContents in the same Profile).

Chrome's `Profile` is equivalent to Electron's `Session`, both of them inherit from `content::BrowserContext`.